packages/dev/core/src/Meshes/Node/Blocks/mappingBlock.ts

```diff
@@ -85,7 +93,8 @@ export class MappingBlock extends NodeGeometryBlock {
         const tempDirection = Vector3.Zero();
 
         const func = (state: NodeGeometryBuildState) => {
-            const position = this.position.getConnectedValue(state) as Vector3;
+            const position = (this.position.getConnectedValue(state) as Vector3) || Vector3.Zero();
+            const normal = (this.normal.getConnectedValue(state) as Vector3) || Vector3.Zero();
             const center = this.center.getConnectedValue(state) as Vector3;

```

Comment:
Why center does not have || Vector3.Zero()

Reply:
it has a default value already as it is an optional